### ISSUE TITLE
Ray does not work with most recent version of redis

### ISSUE THREAD

**Issue by devin-petersohn** [ray-project/ray#3334]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Multiple
- **Ray installed from (source or binary)**: source (current master)
- **Ray version**: - 
- **Python version**: Multiple
- **Exact command to reproduce**: `ray.init()`

### Describe the problem

Multiple people have installed Modin and are receiving this error. https://github.com/modin-project/modin/issues/259

`redis.exceptions.DataError: Invalid input of type: 'NoneType'. Convert to a byte, string or number first.`

I was able to reproduce by running
```
pip install -U redis
```
And then
```python
import ray
ray.init()
```
Redis recently released. I believe we need to pin Ray to a specific version of the library dependencies for compatibility purposes.

**Comment by simon-mo** [comment]
I can re-produce this. A temporary fix will be:
```
pip install -U redis==2.10.6
```

**Comment by devin-petersohn** [comment]
I removed the release-blocker tag since we have pinned Ray to version 2.10.6. We can investigate upgrading redis after the release.

**Comment by richardliaw** [comment]
This issue has come up multiple times; should we pin?

**Comment by robertnishihara** [comment]
@richardliaw, we already have in https://github.com/ray-project/ray/pull/3333.

I think the next step is to just fix the issue.

**Comment by robertnishihara** [comment]
I'll take a quick look at it now.

**Comment by devin-petersohn** [comment]
Closed via #3379
